internal/rootcoord/root_coord.go

```diff
@@ -2364,13 +2369,15 @@ func (c *Core) ReportImport(ctx context.Context, ir *rootcoordpb.ImportResult) (
 
 	// Reverse look up collection name on collection ID.
 	var colName string
-	c.MetaTable.ddLock.RLock()
-	defer c.MetaTable.ddLock.RUnlock()
-	for k, v := range c.MetaTable.collName2ID {
-		if v == ti.GetCollectionId() {
-			colName = k
+	func() {
+		c.MetaTable.ddLock.RLock()
+		defer c.MetaTable.ddLock.RUnlock()
+		for k, v := range c.MetaTable.collName2ID {
+			if v == ti.GetCollectionId() {
+				colName = k
+			}
 		}
```

Comment:
Maybe use `c.MetaTable.GetCollectionByID(ti.GetCollectionId(), 0)` will do the trick?